src/wallet/qi-hdwallet.ts

```diff
@@ -314,4 +323,25 @@ export class QiHDWallet extends AbstractHDWallet {
 		const gapChangeAddresses = this._gapChangeAddresses.filter((addressInfo) => addressInfo.zone === zone);
 		return gapChangeAddresses;
 	}
+
+	public async serialize(): Promise<SerializedQiHDWallet> {
+		const hdwalletSerialized = await super.serialize();
+		return {
+			outpoints: this._outpoints,
+			changeAddresses: Array.from(this._changeAddresses.values()),
+			gapAddresses: this._gapAddresses,
+			gapChangeAddresses: this._gapChangeAddresses,
+			...hdwalletSerialized,
+		};
+	}
+
+	public static async deserialize(data: SerializedQiHDWallet): Promise<QiHDWallet> {
+		const wallet = await super.deserialize<QiHDWallet>(data) as QiHDWallet;
+		wallet._outpoints = data.outpoints;
+		wallet._changeAddresses = new Map(data.changeAddresses.map((addressInfo) => [addressInfo.address, addressInfo]));
+		wallet._gapAddresses = data.gapAddresses;
+		wallet._gapChangeAddresses = data.gapChangeAddresses;
+		return wallet;
+
+	}
```

Comment:
We should do some sort of validation on the outpoints and addresses before assuming they are correct and assigning them to instance vars